libpcap/inject.go

```diff
@@ -0,0 +1,168 @@
+package libpcap
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/asm"
+	"github.com/cloudflare/cbpfc"
+)
+
+/*
+Steps:
+1. Find the injection position, which is the bpf_printk call
+2. Make some necessary preparations for the injection
+3. Compile the filter expression into ebpf instructions
+4. Inject the instructions
+*/
+func InjectFilter(program *ebpf.ProgramSpec, filterExpr string) (err error) {
+	if filterExpr == "" {
```

Comment:
If empty filter, then we will end up calling the bpf_printk?

Reply:
Will revise, thanks for pointing out 👍